degirum_tools/inference_support.py

```diff
@@ -253,16 +261,22 @@ def annotate_video(
         else:
             stream = stack.enter_context(open_video_stream(video_source_id))
 
-        w, h, fps = get_video_stream_properties(stream)
+        w, h, video_fps = get_video_stream_properties(stream)
+
+        # Overwrite the video stream's FPS if the fps argument is set.
+        if fps:
+            video_fps = fps
 
         writer = stack.enter_context(
-            open_video_writer(str(output_video_path), w, h, fps)
+            open_video_writer(str(output_video_path), w, h, video_fps)
         )
 
         if show_progress:
             progress = Progress(int(stream.get(cv2.CAP_PROP_FRAME_COUNT)))
 
-        for res in model.predict_batch(video_source(stream)):
+        decimate_frames = False if isinstance(video_source_id, int) else True
```

Comment:
1. `decimate_frames = not isinstance(video_source_id, int)` is much more readable.
2. Live video streams can also be RTSP cameras, so this condition is technically not correct.
3. It will be much cleaner API, if you will remove this decimate parameter from API at all and will deduce it inside `video_source()` function.

According to chatGPT:

cap.get(cv2.CAP_PROP_FRAME_COUNT) retrieves the number of frames in the video file. If it's a live camera feed, this value would be 0.

So, in order to deduce, which FPS control mode to use (decimation or throttling) you may analyze `cv2.CAP_PROP_FRAME_COUNT` inside `video_source()`: if it is zero, then you have live/realtime stream and you have to use throttling. Otherwise, use decimation.

Reply:
Thanks, I didn't see that, I moved everything to video_source now.